src/dash-table/handlers/cellEvents.ts

```diff
@@ -98,9 +99,15 @@ export const handleChange = (propsFn: () => ICellFactoryProps, idx: number, i: n
         return;
     }
 
+    const result = reconcile(value, c);
+
+    if (!result.success) {
+        return;
+    }
```

Comment:
Make sure the input data can be reconciled with the destination column, otherwise skip